app/models/location.rb

```diff
@@ -1,5 +1,37 @@
+require 'geokit'
+
 class Location < ActiveRecord::Base
   belongs_to :user
 
   validates :user, presence: true
+
+  after_save :reindex_dishes
+
+  def reindex_dishes
+    user.dishes.each { |dish| dish.reindex }
+  end
+
+  # only update latlng when moved distance > 0.5 mile
+  # this is to reduce the cost whenever location is changed, have to reindex dishes (ElasticSearch)
+  def update_latlng(new_lat, new_lng)
+    new_lat = new_lat.try(:to_f) if !new_lat.blank?
+    new_lng = new_lng.try(:to_f) if !new_lng.blank?
+
+    # calculate moved distance
```

Comment:
Try to introduce method and variable names that make the comments obsolete. Comments may become stale over time, and describe behaviour that is no longer true. This is less likely to happen with variable and method names.

For example, lines 20 - 26, without comments, and with names that explain the purpose of the code:

```ruby
    # calculate moved distance
    dist = distance_from new_lat, new_lng

    # only when location change is noticeable
    if dist and dist > 0.5 # 1 mile
      update_attributes(lat: new_lat, lng: new_lng) 
    end
```

becomes:

```ruby
    update_attributes(lat: new_lat, lng: new_lng) if notable_distance_from?(new_lat, new_lng)
```

Think about hiding the `0.5 # 1 mile` value and comment behind a well-named constant. 

Its a little confusing as to what the `0.5` is. It could be a half-mile or in kilometers, or something else. Consider including the units in the name, e.g. `NOTABLE_DISTANCE_IN_THINGUMMIES = 0.5`, replacing `THINGUMMIES` with the correct unit.